### PR TITLE
[collada-dom] Add REMOVE_RECURSE to solve DLLs error path

### LINKED ISSUE #16720: [collada-dom:x64-windows-static] build failure
**Host Environment**
- OS: Windows 10
- Compiler: VS 2019

**To Reproduce**
`vcpkg install collada-dom:x64-windows-static`

**Failure logs** 

```
Installing package uriparser[core]:x64-windows-static... done
Elapsed time for package uriparser:x64-windows-static: 138.4 ms
Starting package 6/6: collada-dom:x64-windows-static
Building package collada-dom[core]:x64-windows-static...
-- Using cached C:/vcpkg/downloads/rdiankov-collada-dom-d37ae7532e350b87c88712e9f6ab4b1f440d20cd.tar.gz
-- Cleaning sources at C:/vcpkg/buildtrees/collada-dom/src/1f440d20cd-abc3ed99e3.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/vcpkg/downloads/rdiankov-collada-dom-d37ae7532e350b87c88712e9f6ab4b1f440d20cd.tar.gz
-- Applying patch vs-version-detection.patch
-- Applying patch use-uriparser.patch
-- Applying patch use-vcpkg-minizip.patch
-- Using source at C:/vcpkg/buildtrees/collada-dom/src/1f440d20cd-abc3ed99e3.clean
-- Configuring x64-windows-static
-- Building x64-windows-static-dbg
-- Building x64-windows-static-rel
-- Installing: C:/vcpkg/packages/collada-dom_x64-windows-static/share/collada-dom/copyright
-- Performing post-build validation
DLLs should not be present in a static build, but the following DLLs were found:

    C:/vcpkg/packages/collada-dom_x64-windows-static/bin/collada-dom2.5-dp-vc140-mt.dll
    C:/vcpkg/packages/collada-dom_x64-windows-static/debug/bin/collada-dom2.5-dp-vc140-mt.dll

There should be no bin\ directory in a static build, but C:\vcpkg\packages\collada-dom_x64-windows-static\bin is present.
There should be no debug\bin\ directory in a static build, but C:\vcpkg\packages\collada-dom_x64-windows-static\debug\bin is present.
If the creation of bin\ and/or debug\bin\ cannot be disabled, use this in the portfile to remove them

    if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
        file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
    endif()

Found 2 error(s). Please correct the portfile:
    C:\vcpkg\ports\collada-dom\portfile.cmake
-- Performing post-build validation done
Error: Building package collada-dom:x64-windows-static failed with: POST_BUILD_CHECKS_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: collada-dom:x64-windows-static
  Vcpkg version: 2021-01-13-d67989bce1043b98092ac45996a8230a059a2d7e
```

**Additional context** 

Only the static version fails, `vcpkg install collada-dom:x64-windows` installs successfully.

### PR DESCRIPTION
**Pull request by JonLiu1993**
**Describe the pull request**

- What does your PR fix? Fixes #16720 
When install colladda-dom:x64-windows-static ，The error shows that DLL files are generated in the bin directory during static install, use REMOVE_RECURSE in the portfile to remove them,

Note: no feature need to test

### CODE REVIEW

File: ports/collada-dom/CONTROL [hunk 1]
```diff
@@ -1,4 +1,6 @@
 Source: collada-dom
 Version: 2.5.0-3
+Homepage: https://github.com/mongodb/mongo-cxx-driver
+Port-Version: 1
```
Comment (NancyLi1013) [inline ports/collada-dom/CONTROL:4]:
```suggestion
Port-Version: 4
```

File: ports/collada-dom/CONTROL [hunk 2]
```diff
@@ -1,4 +1,6 @@
 Source: collada-dom
 Version: 2.5.0-3
```
Comment (NancyLi1013) [inline ports/collada-dom/CONTROL:2]:
```suggestion
Version: 2.5.0
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]
Please also update the patches format.

File: scripts/ci.baseline.txt [hunk 3]
```diff
@@ -137,7 +137,6 @@ cmcstl2:x86-windows        = skip
 coin:arm64-windows=fail
 coin:arm-uwp=fail
 coin:x64-uwp=fail
-collada-dom:x64-windows-static=fail
 collada-dom:x64-windows-static-md=fail
```
Comment (NancyLi1013) [inline scripts/ci.baseline.txt:140]:
```suggestion
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
collada-dom only supports shared library linkage: https://github.com/rdiankov/collada-dom/blob/master/dom/CMakeLists.txt#L31.

We either need to patch out the `SHARED` keyword from the add_library() call or we need to add `vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)` at the top of this port.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks!

**Pull request merged**